crypto/fipsmodule/aes/mode_wrappers.c

```diff
@@ -138,6 +138,15 @@ void AES_cfb128_encrypt(const uint8_t *in, uint8_t *out, size_t length,
   *num = (int)num_u;
 }
 
+#if defined(AES_XTS_X86_64_AVX512)
+int crypto_xts_avx512_enabled(void) {
```

Comment:
I'd prefer if this function is located with the other capability checking in `cpucap/internal.h` as we have for specific ARM platforms in https://github.com/aws/aws-lc/blob/e2818db24344741d68f7011b060ca10cc626e3e6/crypto/fipsmodule/cpucap/internal.h#L199-L216 (but in the x86_64 section).
I don't think it would need to be enclosed in a macro that checks for `MY_ASSEMBLER_IS_TOO_OLD_FOR_512AVX`, since it will be invoked within that check.

Reply:
looking at how we do the checks for AES, I think it's better to just move the function to `aes/internal.h`. For example:
- https://github.com/aws/aws-lc/blob/main/crypto/fipsmodule/aes/internal.h#L47
- https://github.com/aws/aws-lc/blob/main/crypto/fipsmodule/aes/internal.h#L52